CoreFramework/NUnitTestSetup/NUnitTestSetup.cs

```diff
@@ -0,0 +1,64 @@
+﻿using NUnit.Framework;
+using NUnit.Framework.Interfaces;
+using OpenQA.Selenium;
+using CoreFramework.DriverCore;
+using AventStack.ExtentReports;
+using CoreFramework.Reporter;
+
+namespace CoreFramework.NUnitTestSetup;
+
+public class NUnitTestSetup
+{
+    // Check why [SetUp] uses InitDriver
+    // Check Add Project Preference 
+    protected IWebDriver? Driver;
+    public WebDriverAction? DriverBaseAction;
+    protected ExtentReports? _extentReport;
+
+
+    [OneTimeSetUp]
+    public void OneTimeSetUp()
+    {
+        HtmlReport.CreateReport();
+
+        // Without this, a suite will not be created
+        // Assign metadata here
+        HtmlReport.CreateTest(TestContext.CurrentContext.Test.ClassName).
+            AssignAuthor("Hong_Anh_Pham").AssignDevice("PC").AssignCategory("Phase2_TestProject");
+    }
+
+    [SetUp]
+    public void SetUp()
+    {
+        // Pass before initialization => Null
+        // Need a parent obj (Test) before creating child objs (Node/Case)
+        HtmlReport.CreateNode(TestContext.CurrentContext.Test.ClassName, 
+            TestContext.CurrentContext.Test.Name);
+        WebDriverManager.InitDriver("chrome", 1920, 1080);
+        Driver = WebDriverManager.GetCurrentDriver();
+        DriverBaseAction = new WebDriverAction(Driver);
```

Comment:
new WebDriverAction(Driver) in correct param. You should define the URL instead

Reply:
So far I am still struggling with the Driver.Url = baseUrl in WebDriverAction. It seems like that line blocks my code from executing. Here is my temporary fix with SetUp in WebTestSetup by commenting out the Driver.Url = baseUrl , and it works fine for me. Can you make an example of how I can use the baseUrl param? Thanks!

![image](https://user-images.githubusercontent.com/79431746/204551891-8762fdad-cd55-4805-bc2d-8a74d94ea0f4.png)
---
![image](https://user-images.githubusercontent.com/79431746/204552499-6d18fd07-ce76-4de3-acbf-da944dc4e068.png)